GhSA/Helpers/PostHog.cs

```diff
@@ -7,25 +7,70 @@
 using System.Threading.Tasks;
 using Grasshopper.Kernel;
 using Newtonsoft.Json;
+using static GsaGH.Helpers.PostHogUser;
 
 namespace GsaGH.Helpers
 {
+  public static class PostHogUser
+  {
+    public class User
+    {
+      public string email { get; set; }
+      public string userName { get; set; }
+    }
+    private static User thisUser = null;
+    public static User CurrentUser 
+    { 
+      get
+      {
+        if (thisUser == null)
+        {
+          thisUser = new User();
```

Comment:
this.thisUser?

I think what you are trying to do is called a 'singleton': https://www.google.com/search?q=singleton